packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -463,44 +512,22 @@ export default class GoogleCalendarService implements Calendar {
 
     try {
       const calsIds = await getCalIds();
-      const originalStartDate = dayjs(dateFrom);
-      const originalEndDate = dayjs(dateTo);
-      const diff = originalEndDate.diff(originalStartDate, "days");
-
-      // /freebusy from google api only allows a date range of 90 days
-      if (diff <= 90) {
-        const freeBusyData = await this.getCacheOrFetchAvailability({
-          timeMin: dateFrom,
-          timeMax: dateTo,
-          items: calsIds.map((id) => ({ id })),
+      const freeBusyData = await this.getCacheOrFetchAvailability({
+        timeMin: dateFrom,
+        timeMax: dateTo,
+        items: calsIds.map((id) => ({ id })),
+      });
+      if (!freeBusyData?.calendars) throw new Error("No response from google calendar");
+      const result = Object.values(freeBusyData.calendars).reduce((c, i) => {
+        i.busy?.forEach((busyTime) => {
+          c.push({
+            start: busyTime.start || "",
+            end: busyTime.end || "",
+          });
         });
-        if (!freeBusyData) throw new Error("No response from google calendar");
-
-        return freeBusyData;
-      } else {
-        const busyData = [];
-
-        const loopsNumber = Math.ceil(diff / 90);
-
-        let startDate = originalStartDate;
-        let endDate = originalStartDate.add(90, "days");
-
-        for (let i = 0; i < loopsNumber; i++) {
-          if (endDate.isAfter(originalEndDate)) endDate = originalEndDate;
-
-          busyData.push(
-            ...((await this.getCacheOrFetchAvailability({
-              timeMin: startDate.format(),
-              timeMax: endDate.format(),
-              items: calsIds.map((id) => ({ id })),
-            })) || [])
-          );
-
-          startDate = endDate.add(1, "minutes");
-          endDate = startDate.add(90, "days");
-        }
-        return busyData;
-      }
+        return c;
+      }, [] as Prisma.PromiseReturnType<CalendarService["getAvailability"]>);
```

Comment:
just a suggestion, up to you,  I feel like this is a case where using  array methods makes it more complicated and hard to read than necessary

```ts
const calendars = Object.values(freeBusyData.calendars);
const result = [];
for (const calendar of calendars) {
  const busyTimes = calendar.busy ?? [];
  for (const busyTime of busyTimes) {
    result.push({
      start: busyTime.start || "",
      end: busyTime.end || "",
    });
  }
}```